server/spec/subscription_resource_spec.rb

```diff
@@ -48,4 +48,37 @@
           e.to_s.eql? "bad URI(is not URI?): pools/{pool_id}/cert"
       end
   end
+
+
+  it 'subscriptions derived/provided products should return empty array when null' do
+
+      # Product does not have derived/provided products
+      pool = create_pool_and_subscription(@owner['key'], @some_product.id, 2,
```

Comment:
Please don't use the deprecated methods `create_pool_and_subscription` and `delete_pool_and_subscription`.
Instead, use either:
a) `@cp.create_pool()` (and add the subscription_id / upstream_pool_id parameters when you need to simulate the fact that the pool was the result of a manifest import), and `@cp.delete_pool` when you want the test to not be specifically tied to hosted or standalone mode (and run on both), or
b) `create_upstream_subscription` and  `delete_upstream_subscription`, both followed by a call to org_refresh.

Also, this will probably solve the current spec test failures in hosted builds.

Reply:
Ack. Updated code to use new methods.